### PR TITLE
Add data selector caching

### PR DESCRIPTION
**Pull request by alexandrudanpop**
Fixes OPS-1839

## Additional Notes
- When we open data selector on a step we only fetch data for steps that are not in the cache
- On test step / test trigger we store the output in the cache
- Expanded/collapsed is now managed by the parent and stored in the cache
- we don't handle the case of testing steps in the same flow in other tabs
- we might also use this cache in the future when showing step output

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

Deploying it to UX

### CODE REVIEW

File: packages/server/api/src/app/flags/dev-flags.service.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ async function getAll(): Promise<Flag[]> {
   flags = [
     {
       id: FlagId.USE_NEW_EXTERNAL_TESTDATA,
-      value: false,
+      value: true,
```
Comment (alexandrudanpop) [inline packages/server/api/src/app/flags/dev-flags.service.ts:23]: to reset

File: packages/react-ui/src/app/features/builder/update-flow-version.ts [hunk 2]
```diff
@@ -24,13 +25,13 @@ export const updateFlowVersion = (
...
   const newFlowVersion = flowHelper.apply(state.flowVersion, operation);
-  if (
-    operation.type === FlowOperationType.DELETE_ACTION &&
-    operation.request.name === state.selectedStep
-  ) {
-    set({ selectedStep: undefined });
-    set({ rightSidebar: RightSideBarType.NONE });
-    deleteChatRequest(state.flowVersion, operation.request.name);
+  if (operation.type === FlowOperationType.DELETE_ACTION) {
+    stepTestOutputCache.clearStep(operation.request.name);
```
Comment (alexandrudanpop) [inline packages/react-ui/src/app/features/builder/update-flow-version.ts:29]: cleanup cache for deleted step

File: packages/react-ui/src/app/features/builder/tests/update-flow-version.test.ts [hunk 3]
```diff
@@ -33,9 +62,56 @@ describe('updateFlowVersion', () => {
             },
           },
         },
+        updatedBy: null,
+        valid: false,
+        state: FlowVersionState.DRAFT,
       },
       selectedStep: 'step_1',
       rightSidebar: RightSideBarType.BLOCK_SETTINGS,
+      readonly: false,
```
Comment (cezudas) [inline packages/react-ui/src/app/features/builder/tests/update-flow-version.test.ts:71]: Any reason why the exhaustive list of BuilderState properties is needed for the mock?

**Review by cezudas — COMMENTED** [review]
great work, I only left a couple of minor questions for clarification purposes!

File: packages/react-ui/src/app/features/builder/data-selector/use-selector-data.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    });
+    forceRerender((v) => v + 1);
+    return stepTestOutput;
+  };
+
+  const { isLoading } = useQuery({
+    queryKey: [QueryKeys.dataSelectorStepTestOutput, flowVersionId, ...stepIds],
+    queryFn: async () => {
+      if (initialLoad) {
+        setInitialLoad(false);
```
[the remark this answers is not in the document]
Comment (cezudas) [reply packages/react-ui/src/app/features/builder/data-selector/use-selector-data.ts:42]: it is what it is then
